### PR TITLE
Final changes

### PR DESCRIPTION
**Pull request by chelseajohn**
These include the Synchronization update, Saving before and after intermediate models as well the header for log files

### CODE REVIEW

File: DLplatform/coordinator.py [hunk 1]
```diff
@@ -236,12 +236,19 @@ def onMessageReceived(self, routing_key, exchange, body):
             # we send around the initial parameters only when all the expected nodes are there
             # in case when parameter is not set, it is equal to 0 - so every new node will satisfy the condition
             if len(self._waitingNodes) >= self._minStartNodes:
+                for id in self._waitingNodes:
```
Comment (link-er) [inline DLplatform/coordinator.py:239]: can you please update it in a way that is currently in the master branch? there were some changes that this update will override. I guess overall you were not changing coordinator.py? so it should be equivalent to the original

File: DLplatform/worker.py [hunk 2]
```diff
@@ -235,7 +235,8 @@ def onCommunicatorMessageReceived(self,  routing_key, exchange, body):
             message = pickle.loads(body)
             param = message['param']
             flags = message['flags']
-            self._learner.setModel(param, flags)
+            self._learner.setModel(param, flags) # before intermediate models are logged in setModel() function
```
Comment (link-er) [inline DLplatform/worker.py:238]: but this logging (before) is not there? I do not see it in the commited files

File: DLplatform/coordinator.py (hunk 1 above)
Comment (chelseajohn) [reply DLplatform/coordinator.py:239]: I have merged in the recent commits from the Fraunhofer master branch. So there wont be any conflicts

**Comment by link-er** [comment]
merged another version

**Pull request closed without merging**